devel/boost/Portfile

```diff
@@ -117,6 +117,11 @@ configure.args      --without-libraries=python \
                     --without-libraries=mpi \
                     --with-icu=${prefix}
 
+# boost build scripts default to clang on darwin
+if {[string match *gcc* ${configure.cc}]} {
```

Comment:
hmm ... Boost is c++ ... so shouldn't we check for `${configure.cxx}`?

Reply:
You would think so! But the cxx gcc compiler identifies itself as “g++-mp-N”, so matching against “gcc” doesn’t work. I didn’t want to match “clang++-anything”... and so used configure.cc for (hopefully) clarity and specificity... 

What do you think?

Reply:
Maybe just configure.compiler might work and be clearer?

Reply:
Ah yes ... interesting issue there! Right so MP clang would be `clang++-mp-*`, while GCC's would be `g++-mp-*` ... different `*` but whatever. Not an easy match.

Primary downside is that one could technically do something like "sudo port install boost configure.cc=gcc-mp-6" and then this tweak would happen but the actual CXX compiler would be whatever was selected by port ... some clang for most people.

Is there no better way to do this?

Reply:
If our only 2 compiler options are clang and gcc, then you could always do something like
`if{![string match *clang* ${configure.cxx}]} {`
maybe?

Reply:
Couldn't you match against `${configure.compiler}`, which should be an identifier for the chosen compiler suite?

Reply:
I’m fine with all three of these. Perhaps using configure.compiler might be the most clear in two years 😀